### PR TITLE
Implements MXMNet Model Class

### PR DESCRIPTION
**Pull request by riya-singh28**
## Description

Implements MXMNet Class and regression test.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by shreyasvinaya** [comment]
@riya-singh28 Can you please rebase the PR, we can fix the errors that pop up once we are on the latest version of master branch

File: requirements/env_dqc.yml [hunk 1]
```diff
@@ -18,3 +18,5 @@ dependencies:
    - PyYAML
    - yamlloader
    - pyscf[all]
+   - -f https://data.pyg.org/whl/torch-2.1.0+cpu.html
```
Comment (shreyasvinaya) [inline requirements/env_dqc.yml:21]: do we need to add torch_cluster to dqc?

File: deepchem/models/torch_models/mxmnet.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        This method ensures that the node embeddings are initialized with random values, promoting diversity in the
+        initial node representations.
+        """
+
+        stdv: float = math.sqrt(3)
+        self.embeddings.data.uniform_(-stdv, stdv)
+
+    def indices(
+        self, edge_index: torch.Tensor, num_nodes: int
+    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor,
```
Comment (aaronrockmenezes) [inline deepchem/models/torch_models/mxmnet.py:153]: Why do we have some many torch tensors? Can you put a comment for what the values represent in the code?

**Review by aaronrockmenezes — CHANGES_REQUESTED** [review]

File: deepchem/models/torch_models/mxmnet.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+    -------
+    >>> import deepchem as dc
+    >>> import os
+    >>> import tempfile
+    >>> import torch
+    >>> from torch_geometric.data import Data, Batch
+    >>> from deepchem.feat.molecule_featurizers import MXMNetFeaturizer
+    >>> QM9_TASKS = ["mu", "alpha", "homo", "lumo", "gap", "r2", "zpve", "cv", "u0", "u298",
+    ...              "h298", "g298"]
+    >>> device = 'cpu'
```
Comment (aaronrockmenezes) [inline deepchem/models/torch_models/mxmnet.py:36]: Why are we setting device to cpu in the usage example?

**Review by aaronrockmenezes — CHANGES_REQUESTED** [review]

File: deepchem/models/torch_models/mxmnet.py [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+        num_radial: int, default 6
+            The number of radial basis functions.
+        envelope_exponent: int, default 5
+            The exponent for the envelope function.
+        activation_fn: Union[Callable, str], default 'silu'
+            The activation function name.
+        n_tasks: int, default 1
+            The number of prediction tasks. Only single Task regression is supported currently.
+        """
+        seed = 123
```
Comment (shreyasvinaya) [inline deepchem/models/torch_models/mxmnet.py:94]: is there any reason we are seeding the main class?

**Comment by shreyasvinaya** [comment]
@riya-singh28 Torch-cluster seems to be breaking the macos install script, could you please have a look at it

File: requirements/env_dqc.yml (hunk 1 above)
Comment (riya-singh28) [reply requirements/env_dqc.yml]: I have removed it. It was not needed.

File: deepchem/models/torch_models/mxmnet.py (hunk 2 above)
Comment (riya-singh28) [reply deepchem/models/torch_models/mxmnet.py:153]: Those are node indices for various angle calculations.

File: deepchem/models/torch_models/mxmnet.py (hunk 3 above)
Comment (riya-singh28) [reply deepchem/models/torch_models/mxmnet.py:36]: This was just for testing, I'll remove it. It won't cause any change.

**Comment by riya-singh28** [comment]
Made a new PR - https://github.com/deepchem/deepchem/pull/3970

**Pull request closed without merging**